packages/svelte/src/compiler/phases/2-analyze/validation.js

```diff
@@ -622,6 +622,11 @@ const validation = {
 	SnippetBlock(node, context) {
 		validate_block_not_empty(node.body, context);
 
+		const old_parent = context.state.parent_element;
+		context.state.parent_element = 'snippet';
+		context.next(context.state);
+		context.state.parent_element = old_parent;
```

Comment:
we want to avoid mutating state

```suggestion
		context.next({ ...context.state, parent_element: 'snippet' });
```

Reply:
Gotcha ..will keep in mind for the future